x-pack/plugins/ml/public/application/explorer/anomaly_charts_state_service.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, combineLatest, Observable, of, Subscription } from 'rxjs';
+import { distinctUntilChanged, map, skipWhile, switchMap } from 'rxjs/operators';
+import { StateService } from '../services/state_service';
+import type { AnomalyExplorerCommonStateService } from './anomaly_explorer_common_state';
+import type { AnomalyTimelineStateService } from './anomaly_timeline_state_service';
+import {
+  ExplorerChartsData,
+  getDefaultChartsData,
+} from './explorer_charts/explorer_charts_container_service';
+import { AnomalyExplorerChartsService } from '../services/anomaly_explorer_charts_service';
+import { getSelectionInfluencers } from './explorer_utils';
+import type { PageUrlStateService } from '../util/url_state';
+import type { TableSeverity } from '../components/controls/select_severity/select_severity';
+import { AnomalyExplorerUrlStateService } from './hooks/use_explorer_url_state';
+
+export class AnomalyChartsStateService extends StateService {
+  private _isChartsDataLoading$ = new BehaviorSubject<boolean>(false);
+  private _chartsData$ = new BehaviorSubject<ExplorerChartsData>(getDefaultChartsData());
+  private _showCharts$ = new BehaviorSubject<boolean>(true);
+
+  constructor(
+    private _anomalyExplorerCommonStateService: AnomalyExplorerCommonStateService,
+    private _anomalyTimelineStateServices: AnomalyTimelineStateService,
+    private _anomalyExplorerChartsService: AnomalyExplorerChartsService,
+    private _anomalyExplorerUrlStateService: AnomalyExplorerUrlStateService,
+    private _tableSeverityState: PageUrlStateService<TableSeverity>
+  ) {
+    super();
+    this._init();
+  }
+
+  protected _initSubscribtions(): Subscription {
```

Comment:
Nit - typo, should be `_initSubscriptions`

Reply:
fixed in https://github.com/elastic/kibana/pull/128165/commits/1d6b8919637c1a216eb5bce735ae07c8631dfadf